ExampleMod/Content/Prefixes/ExamplePrefix.cs

```diff
@@ -0,0 +1,67 @@
+using ExampleMod.Items;
+using Terraria;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Prefixes
+{
+    public class ExamplePrefix : ModPrefix
+    {
+        private readonly byte _power;
+
+        // See documentation for vanilla weights and more information.
+        // Note: a weight of 0f might still be rolled. See CanRoll to exclude prefixes.
+        // Note: if you use PrefixCategory.Custom, actually use ModItem.ChoosePrefix instead.
+        public override float RollChance(Item item)
+            => 5f;
+
+        // Determines if it can roll at all.
+        // Use this to control if a prefix can be rolled or not.
+        public override bool CanRoll(Item item)
+            => true;
+
+        // Change your category this way, defaults to PrefixCategory.Custom. Affects which items can get this prefix.
+        public override PrefixCategory Category
+            => PrefixCategory.AnyWeapon;
+
+        // This is needed for the tModLoader to attempt Autoloading.
+        public ExamplePrefix()
+        {
+        }
+
+        // The actual constructor which is used in the Autoload.
+        public ExamplePrefix(byte power)
+        {
+            _power = power;
+        }
+
+        // Allow multiple prefix autoloading this way (Different versions of the same prefix).
+        public override bool Autoload(ref string name)
+        {
+            if (base.Autoload(ref name))
+            {
+                mod.AddPrefix("Awesome", new ExamplePrefix(1));
+                mod.AddPrefix("ReallyAwesome", new ExamplePrefix(2));
+            }
+
+            return false;
+        }
```

Comment:
This looks like 1.3 code. Have you tested this in 1.4?

Reply:
adding prefixes like this requires ModSystem.Load or Mod.Load now and some additional code for the prefix constructor (Name property), here is an example:
Where it's called: https://github.com/direwolf420/ClickerClass/blob/1.4_experiment/ClickerClass.cs#L26
Prefix code and registering (see AddContent): https://github.com/direwolf420/ClickerClass/blob/1.4_experiment/Prefixes/ClickerPrefix.cs#L8

Reply:
I saw, working on it at the moment (Apparently testing would require me to update Visual Studio, so this takes some time)